jvm-statistics/src/main/java/com/criteo/hadoop/garmadon/jvm/statistics/NetworkStatistics.java

```diff
@@ -2,86 +2,216 @@
 
 import com.criteo.hadoop.garmadon.jvm.AbstractStatistic;
 import com.criteo.hadoop.garmadon.jvm.StatisticsSink;
-import oshi.SystemInfo;
-import oshi.hardware.NetworkIF;
 
+import java.io.BufferedReader;
+import java.io.File;
+import java.io.IOException;
+import java.io.InputStreamReader;
+import java.io.RandomAccessFile;
+import java.lang.management.ManagementFactory;
 import java.util.ArrayList;
+import java.util.Collections;
+import java.util.HashMap;
 import java.util.List;
+import java.util.Map;
+import java.util.regex.Matcher;
+import java.util.regex.Pattern;
 
 class NetworkStatistics extends AbstractStatistic {
     private static final String NETWORK_HEADER = "network";
-    private static final String NETWORK_RECV_SUFFIX = "_rx";
-    private static final String NETWORK_SENT_SUFFIX = "_tx";
-    private static final String NETWORK_PKT_RECV_SUFFIX = "_pktrx";
-    private static final String NETWORK_PKT_SENT_SUFFIX = "_pkttx";
-    private static final String NETWORK_ERRORS_IN_SUFFIX = "_errin";
-    private static final String NETWORK_ERRORS_OUT_SUFFIX = "_errout";
+    private static final String NETWORK_PROCESS_RECV = "processrx";
+    private static final String NETWORK_PROCESS_SENT = "processtx";
+    private static final Pattern SS_SOCKET_LINE_1 = Pattern.compile("ESTAB\\s+\\d+\\s+\\d+\\s+(\\d+\\.\\d+\\.\\d+.\\d+:\\d+)\\s+(\\d+\\.\\d+\\.\\d+.\\d+:\\d+)\\s+users:\\(\\(.*,pid=(\\d+),.*\\)\\)");
+    private static final Pattern SS_SOCKET_LINE_RECEIVED = Pattern.compile("bytes_received:(\\d+)");
+    private static final Pattern SS_SOCKET_LINE_ACKED = Pattern.compile("bytes_acked:(\\d+)");
+    private static final File PROC_FILE = new File("/proc");
 
-    private final List<NicInfo> nics = new ArrayList<>();
+    private final int currentPid;
+    private final Map<String, SocketStats> socketStats = new HashMap<>();
 
-    NetworkStatistics() {
+    public NetworkStatistics() {
         super(NETWORK_HEADER);
-        for (NetworkIF nic : new SystemInfo().getHardware().getNetworkIFs()) {
-            if (nic.getBytesSent() == 0 && nic.getBytesRecv() == 0) {
-                // nothing happens on this nic since boot
-                continue;
-            }
-            nics.add(new NicInfo(nic));
-        }
+        this.currentPid = extractCurrentPid();
```

Comment:
Add check on availibility of ss command otherwise exit fast innerCollect

Please add deployment of iproute package in test/src/main/docker/garmadon/Dockerfile